### ISSUE TITLE
xpub problems

### ISSUE THREAD

**Issue by ChzenChzen** [sbuss/bitmerchant#64]
I took xpub key from ledger nano. I generated btc address After bitmerchant library had imported . Sent small amount btc on this address, but after that I didn't have any new incoming transactions, only old transactions exist which I had done via ledger nano. I used blockchain.info for checking. Can anybody help me?

**Comment by ChzenChzen** [comment]
How to generate segwit addresses from xpub?

**Comment by ZykC** [comment]
Hello ChzenChzen,

I was just trying the same manipulation using nano X and bitmerchant address generation and have the same issue. 

Have you figure out why you don't see incoming transaction on the xpub ?

Thanks in advance !

**Comment by sbuss** [comment]
@ZykC as you can see, this project is more-or-less abandoned. I can attempt to help if you give me a few more details, though.

**Comment by ZykC** [comment]
Thank you for  your answer.
I have a ledger nano X connected to ledger live app. I exported the segwit xpub from the app and I try to use is in python based website, I want to generate an address for each user to deposit fund in the website, I use Bitmerchant: 
> from bitmerchant.wallet import Wallet
> XPUB = "xpub6C...."
> user_id = 100
> master_wallet = Wallet.deserialize(XPUB)
> user_wallet = master_wallet.create_new_address_for_user(user_id)
> user_address = user_wallet.to_address()

This generated me a legacy wallet "1MN......"
I made a small transfer to this address, but unfortunately, I wasn't able to see it in my wallet, I checked in the Blockchain explorer, same thing, nothing.

Then I tried a library found here https://github.com/mcdallas/cryptotools to generate a segwit address from the Xpub, I managed to generate the segwit address "bc1q...." with the following instructions: 
> from btctools import Xpub
> extended = Xpub.decode('xpub6...')
> child = extended/0/0
> user_address = child.key.to_address('P2WPKH')
I tested with a small amount and nothing, I received nothing.
It's the first time I try to integrate HD Wallet to a website, am I missing something ?

Thanks !
